### PR TITLE
Convert from Redux to MobX

### PR DESCRIPTION
**Pull request by fwouts**
This refactors the entire state management architecture of Comet away from Redux Observable / RxJS, towards MobX. It comes with a full suite of unit tests, and it's still overall less verbose :)

Note that further improvements should be made in future PRs. In particular, each UI component should be decoupled from the state.

### CODE REVIEW

**Comment by driesg** [comment]
I'm not feeling very confident in MobX at the moment. Do we have a specialist in the house? I'd be keen to review and use this PR to refresh my memory of it though but that means it'll take a couple days

**Comment by fwouts** [comment]
@driesg Please take a look :)

File: src/jira/loader.ts [hunk 1]
```diff
@@ -1,98 +1,82 @@
...
+    const allTickets = await Promise.all(
+      jiraKeys.map(jiraKey =>
+        this.loadTicket(jiraKey).catch(e => {
+          console.warn(e);
+          return null;
+        })
+      )
+    );
+    return allTickets.reduce<{
+      [jiraKey: string]: JiraTicket;
```
Comment (driesg) [inline src/jira/loader.ts]: this could be made more readable with a type alias

File: src/jira/loader.ts [hunk 2]
```diff
@@ -1,98 +1,82 @@
...
+          .map(r => r.commits)
+          .reduce((acc, curr) => acc.concat(curr), [])
+      )
+      .reduce((acc, curr) => acc.concat(curr), [])
+      .sort((a, b) => {
+        return (
+          new Date(b.authorTimestamp).getTime() -
+          new Date(a.authorTimestamp).getTime()
+        );
+      });
```
Comment (driesg) [inline src/jira/loader.ts]: Maybe add some comments here, explaining how the data is processed

File: src/routing.ts [hunk 3]
```diff
@@ -1,4 +1,44 @@
-export function producePath(
+import { createBrowserHistory, History, Location } from "history";
```
Comment (driesg) [inline src/routing.ts]: use react router? (or delete from package.json)

**Pull request merged**